kong/conf_loader.lua

```diff
@@ -425,6 +426,46 @@ local function check_and_infer(conf)
     end
   end
 
+  -- Validate origins
+  local seen_origins = {}
+
+  for i, v in ipairs(conf.origins) do
+    local from_scheme, from_host_port, to_host_port =
+      v:match("^(https?)://([^=]+:[%d]+)=https?://([^/]+)$")
+
+    if not from_scheme then
+      errors[#errors + 1] = "an origin must be of the form " ..
+                            "'from_scheme://from_host:from_port=" ..
+                            "to_scheme://to_host:to_port', got '" ..
+                            v .. "'"
+
+    else
+      -- Validate 'from'
+      local from_authority, err =
+        utils.format_host(utils.normalize_ip(from_host_port))
```

Comment:
This code now normalized an ipv6 address twice. Once before calling format and once inside the format call.